interp/run.go

```diff
@@ -94,9 +94,13 @@ func (interp *Interpreter) run(n *node, cf *frame) {
 		f = newFrame(cf, len(n.types), interp.runid())
 	}
 	interp.mutex.RLock()
-	f.done = reflect.SelectCase{Dir: reflect.SelectRecv, Chan: reflect.ValueOf(interp.done)}
+	c := reflect.ValueOf(interp.done)
 	interp.mutex.RUnlock()
 
+	f.mutex.RLock()
+	f.done = reflect.SelectCase{Dir: reflect.SelectRecv, Chan: c}
+	f.mutex.RUnlock()
```

Comment:
I'm probably missing the big picture (I need to look at the whole code), but just looking at this PR, it seems to me that we now have RLocks in all the places where we read from f.done, so it seems to be that here, we should have a Lock/Unlock, since we are writing to f.done.

Reply:
Right. a Lock/Unlock is needed here.